include/internal/refcount.h

```diff
@@ -176,11 +173,15 @@ typedef int CRYPTO_REF_COUNT;
 #  define REF_ASSERT_ISNT(i)
 # endif
 
-# ifdef REF_PRINT
-#  define REF_PRINT_COUNT(a, b) \
-        fprintf(stderr, "%p:%4d:%s\n", (void*)b, b->references, a)
+# ifdef REF_DEBUG
+#  define REF_PRINT_EX(text, count, object) \
+    ossl_refcount_print((void *)object, count, text)
+#  define REF_PRINT_COUNT(text, object) \
+    ossl_refcount_print((void *)object, object->references, text)
 # else
+#  define REF_PRINT_EX(a, b)
 #  define REF_PRINT_COUNT(a, b)
 # endif
 
+void ossl_refcount_print(void *object, int count, const char *text);
```

Comment:
How about this?

``` suggestion
# define REF_PRINT(text, count, object) \
    OSSL_TRACE3(REF_COUNT, "%p:%4d:%s\n", (void *)object, (int)refcount, text)
# define REF_PRINT_COUNT(text, object) REF_PRINT((text), (object)->reference, (object))
```

Considering that OSSL_TRACE3() is already conditionally something or nothing depending on configuring with or without `enable-trace`, I don't see a reason why anyone should *have to* define `REF_DEBUG` *as well*

With this, `ossl_refcount_print()` can be thrown away...
```

Reply:
I wanted to make a less invasive change, but will do it.